pkg/cloud/gcp/gpu.go

```diff
@@ -0,0 +1,46 @@
+package gcp
+
+import "strings"
+
+// gpuSKUToGpuLabel defines minimal, explicit mappings for common SKUs.
+var gpuSKUToGpuLabel = map[string]string{
+	"nvidia tesla a100 80gb": "nvidia-a100-80gb",   // A2-Ultra
+	"nvidia a100 80gb":       "nvidia-a100-80gb",
+	"nvidia tesla a100":      "nvidia-tesla-a100",  // A2-HighGPU legacy label
+	"nvidia a100":            "nvidia-tesla-a100",
+	"nvidia l4":              "nvidia-l4",          // G2 nodes
+	"tesla t4":               "nvidia-tesla-t4",
+	"nvidia t4":              "nvidia-tesla-t4",
+	"tesla v100":             "nvidia-tesla-v100",
+	"nvidia v100":            "nvidia-tesla-v100",
+}
+
+// NormalizeGPULabel converts a billing SKU description into the GKE/OpenCost GPU label.
+func NormalizeGPULabel(desc string) string {
+	d := strings.ToLower(desc)
+
+	// Fast path: explicit substring matches first
+	for key, model := range gpuSKUToGpuLabel {
+		if strings.Contains(d, key) {
+			return model
+		}
+	}
+
+	// Fallbacks/cleanup (keeps your current behavior)
+	//  - drop packaging suffixes, unify prefix, handle generic A100 40/80GB
+	g := d
+	g = strings.ReplaceAll(g, "-sxm4", "")
+	g = strings.ReplaceAll(g, "  ", " ")
+	g = strings.TrimSpace(g)
+
+	// If we can spot A100 capacity, decide 80GB vs legacy A100 label
```

Comment:
Hi @mohamedosama113 , This is great, definitely step in right direction: Can u gut check me on something thou?

For product sku description like `Nvidia Tesla P100 GPU running in Melbourne`. The original code would have correctly parsed the label to be `nvidia-tesla-p100`. Ref : https://go.dev/play/p/a_V_zfbdjyf

Can you add that text to the test case and it would return empty string. Correct me if im wrong there?

One last feedback i have is can u just copy paste the existing regex FindStringSubmatch with those consts[which can be moved here to this file] as the fallback! I'm good after that!!!!! TY again!

Reply:
Eventually we would want the map to be the source of truth. The Product sku and label description by GKE wont change unless there is an announcement but THe regex as a fallback for now is good. What do you think?